docs/contributing/testing-guidelines/integration-testing.md

```diff
@@ -0,0 +1,253 @@
+
+Integration tests help to find the following types of errors:
+
+- Incompatible interactions between nodes, such as non-matching topics, different message types, or incompatible QoS settings.
+- Edge cases that were not touched by unit testing, such as a critical timing issue, network communication delays, disk I/O failures, and many other problems that can occur in production environments.
+- Issues that can occur while the system is under high CPU/memory load, such as `malloc` failures. This can be tested using tools like `stress` and `udpreplay` to test the performance of nodes with real data.
+
+With ROS 2, it is possible to program complex autonomous-driving applications with a large number of nodes. Therefore, a lot of effort has been made to provide an integration-test framework that helps developers test the interaction of ROS 2 nodes.
+
+## Integration-test framework
+
+A typical integration-test framework has three parts:
+
+1. A series of executables with arguments that work together and generate outputs.
+2. A series of expected outputs that should match the output of the executables.
+3. A launcher that starts the tests, compares the outputs to the expected outputs, and determines if the test passes.
+
+In Autoware, we use the [launch_testing](https://github.com/ros2/launch/tree/master/launch_testing) framework.
+
+### Smoke tests
+
+Autoware has a dedicated API for smoke testing. To use this framework, in `package.xml` add:
+
+```xml
+<test_depend>autoware_testing</test_depend>
+```
+
+And in `CMakeLists.txt` add:
+
+```cmake
+if(BUILD_TESTING)
+  find_package(ament_lint_auto REQUIRED)
+  ament_lint_auto_find_test_dependencies()
+
+  find_package(autoware_testing REQUIRED)
+  add_smoke_test(${PROJECT_NAME} ${NODE_NAME})
+endif()
+```
+
+Doing so adds smoke tests that ensure that a node can be:
+
+1. Launched with a default parameter file.
+2. Terminated with a standard `SIGTERM` signal.
+
+For the full API documentation, refer to the [package design page](https://github.com/autowarefoundation/autoware.universe/blob/main/common/autoware_testing/design/autoware_testing-design.md).
+
+!!! note
+
+    This API is not suitable for all smoke test cases. 
+    It cannot be used when a specific file location (eg: for a map) is required to be passed to the node, or if some preparation needs to be conducted before node launch.
+    In such cases use the manual solution from the [component test section below](#integration-test-with-a-single-node-component-test).
+
+### Integration test with a single node: component test
+
+The simplest scenario is a single node. In this case, the integration test is commonly referred to as a component test.
+
+To add a component test to an existing node, follow the example of the `lanelet2_map_provider` package that has an executable named `lanelet2_map_provider_exe`.
```

Comment:
If this is a real package, which repository is it in? I searched the AWF org on GitHub and GitLab but couldn't find it. If the package is an existing one, then we should add a link to it.

Reply:
The tests are actually quite different and I am not sure if the one in `fault_injection` is a good example for the documentation.

Reply:
I wrote a new test that I think can be used as a good example: https://github.com/autowarefoundation/autoware.universe/pull/1056